test/lib/ecto_types/schedule_test.exs

```diff
@@ -142,6 +143,52 @@ defmodule Ask.ScheduleTest do
       time = @schedule |> Schedule.next_available_date_time(base)
       assert time == DateTime.from_naive!(~N[2017-10-11 12:00:00], "Etc/UTC")
     end
+
+    test "gets next available time: start date" do
+      # Long before the start date
+      base = DateTime.from_naive!(~N[2017-02-01 00:00:00], "Etc/UTC")
+      time = @schedule |> Schedule.next_available_date_time(base)
+      # The beginning of the first active window after the start date
+      assert time == DateTime.from_naive!(~N[2017-03-05 12:00:00], "Etc/UTC")
+    end
+  end
+
+  describe "last_window_ends_at" do
+    setup do
+      base_schedule = %Ask.Schedule{
+        start_time: ~T[09:00:00],
+        end_time: ~T[18:00:00],
+        day_of_week: %Ask.DayOfWeek{sun: true, wed: true},
+        timezone: "America/Argentina/Buenos_Aires",
+        start_date: ~D[2017-03-03],
+        blocked_days: [~D[2017-10-08]]
+      }
+      {:ok, base_schedule: base_schedule}
+    end
+
+    test "gets last available time: unavaible day", %{base_schedule: base_schedule} do
```

Comment:
typo here: I assume you wanted to write `unavailable` :)

Reply:
I don't think so, but yes, I'll improve the expressiveness of this test too. Thanks!!